### ISSUE TITLE
Accessibility: Associate label with form control for search on table charts with search box

### ISSUE THREAD

**Issue by yousoph** [apache/superset#26175]
On table chart types, the label for the search box form control isn't associated with the search field, making it difficult to understand the relationship for those using a screen reader

#### How to reproduce the bug

1. Create a table chart
2. In the Customize tab, enable the "Search Box" option 
3. Add the chart to a dashboard 
4. While on the dashboard, enable Screen Reader and tab through the dashboard content 

### Expected results
https://www.w3.org/TR/WCAG22/#info-and-relationships
User is able to understand what the search box is associated when using the screen reader 

### Actual results
The label for the search box isn't associated with the form control, so the user is unable to tell what the search box is for when using a screen reader

#### Screenshots

![image](https://github.com/apache/superset/assets/10627051/69079c20-95df-41ed-8637-dd5725d2dfbb)

### Environment

(please complete the following information):

- browser type and version:
- superset version: `master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by APH1997** [comment]
I'll take this one on!

**Comment by yousoph** [comment]
https://github.com/apache/superset/pull/27968
